webapp/src/components/sidebar/ModelLibrary.vue

```diff
@@ -210,11 +212,34 @@ function sortModels(a: ModelMetadata, b: ModelMetadata): number {
 }
 
 
-function deleteModelConfirm(modelID: string){
-  toaster.default(`Click here to confirm the model deletion`, {
-    onClick: () => deleteModel(modelID),
-    duration: 10000,
-  });
+async function deleteModelConfirm(modelID: string){
+  const modelInfo = models.value.find(([id]) => id === modelID);
+  const model = await memory.getModel(modelID);
+  if (!modelInfo) return
+
+  deleteModel(modelID).then(() => {
+    console.log("ici")
+    tempDeletedModel = { id : modelID, model : model, metadata : modelInfo[1]};
+    toaster.info("Model deleted. Click here to undo", { 
+      onClick: () => undoDeleteModel(),
```

Comment:
```suggestion
      onClick:  undoDeleteModel,
```
If you want to be fancy